contrib/drivers/ibm/ibm.go

```diff
@@ -0,0 +1,275 @@
+		 TgtConfDir: defaultTgtConfDir,
+		 UserName: username,
+                 Port: port,
+                 Password: password,
+	}
+	p := config.CONF.OsdsDock.Backends.IBM.ConfigPath
+	if "" == p {
+		p = defaultConfPath
+	}
+	if _, err := Parse(d.conf, p); err != nil {
+		return err
+	}
+	err:= login()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (*Driver) Unset() error { return nil }
+
+// first get the status of spectrumstate. If it is not active just return
+func (d *Driver) CreateVolume(opt *pb.CreateVolumeOpts) (vol *model.VolumeSpec, err error) {
+		 err = d.cli.GetSpectrumScaleStatus()
+		 if err != nil{
+			 log.Error("the GPFS cluster is not active")
+			 return &model.VolumeSpec{}, err
+		 }
+		 // if spectrumscale service is active, create volume(fileset)
+		 log.Info("IBM driver receive create volume request, vr =", opt)
+		 var volName = volumePrefix + opt.GetId()
+		 var volSize = opt.GetSize()
+		 size := strconv.FormatInt(int64(volSize), 10)
+		 if err = d.cli.CreateVolume(volName, size); err != nil {
+	 		return &model.VolumeSpec{}, err
+	 	}
+
+                var mountPoint string
+		mountPoint, err = d.cli.GetSpectrumScaleMountPoint()
+
+		if err != nil{
+			log.Error("not able to find spectrumscale mount point")
+			return &model.VolumeSpec{}, err
+		}
+
+		return &model.VolumeSpec{
+			BaseModel: &model.BaseModel{
+				Id: opt.GetId(),
+			},
+			Name:        opt.GetName(),
+			Size:        opt.GetSize(),
+			Description: opt.GetDescription(),
+			Identifier:  &model.Identifier{DurableName:opt.GetId(),
+				DurableNameFormat: KLvIdFormat,
+			},
+			Metadata: map[string]string{
+				FileSetPath: mountPoint + "/" + volName,
+			},
+		}, nil
+}
+
+// discover the pool from spectrumscale
+func (d *Driver) ListPools() ([]*model.StoragePoolSpec, error) {
+	var mountPoint string
+	mountPoint, stderr := d.cli.GetSpectrumScaleMountPoint()
+	if stderr != nil {
+		return nil, stderr
+	}
+
+	pools, err := d.cli.ListPools(mountPoint)
+	if err != nil {
+		return nil, err
+	}
+	// retrive the all details from spectrumscale pool
+	var pols []*model.StoragePoolSpec
+	for _, pool := range *pools {
+		pol := &model.StoragePoolSpec{
+			BaseModel: &model.BaseModel{
+				Id: uuid.NewV5(uuid.NamespaceOID, pool.UUID).String(),
+			},
+			Name:             pool.Name,
+			TotalCapacity:    pool.TotalCapacity,
+			FreeCapacity:     pool.FreeCapacity,
+			StorageType:      storageType,
+			Extras:           d.conf.Pool[pool.Name].Extras,
+			AvailabilityZone: d.conf.Pool[pool.Name].AvailabilityZone,
+			MultiAttach:      d.conf.Pool[pool.Name].MultiAttach,
+		}
+		if pol.AvailabilityZone == "" {
+			pol.AvailabilityZone = "default"
+		}
+		pols = append(pols, pol)
+	}
+	return pols, nil
+}
+
+// this function is for deleting the spectrumscale volume(fileset)
+func (d *Driver) DeleteVolume(opt *pb.DeleteVolumeOpts) error{
+        fileSetPath:= opt.GetMetadata()[FileSetPath]
+	field := strings.Split(fileSetPath, "/")
```

Comment:
Put comment with a sample filesetpath, so that reader knows why the fourth field is required